common/api/ecschema-locaters.api.md

```diff
@@ -19,38 +20,56 @@ export class FileSchemaKey extends SchemaKey {
     schemaText?: string;
 }
 
-// @beta
+// @alpha
 export abstract class SchemaFileLocater {
     constructor();
     addSchemaSearchPath(schemaPath: string): void;
     addSchemaSearchPaths(schemaPaths: string[]): void;
+    // (undocumented)
+    addSchemaText(schemaPath: string, readSchemaText: DelayedPromise<string | undefined>): Promise<void>;
+    // (undocumented)
+    addSchemaTextSync(schemaPath: string, readSchemaText: DelayedPromise<string | undefined>): void;
     compareSchemaKeyByVersion: (lhs: FileSchemaKey, rhs: FileSchemaKey) => number;
     // (undocumented)
     fileExists(filePath: string): Promise<boolean | undefined>;
     // (undocumented)
     fileExistsSync(filePath: string): boolean | undefined;
-    protected findEligibleSchemaKeys(desiredKey: SchemaKey, matchType: SchemaMatchType, format: string): FileSchemaKey[];
+    protected findEligibleSchemaKeys(desiredKey: SchemaKey, matchType: SchemaMatchType, format: string): Promise<FileSchemaKey[]>;
+    protected findEligibleSchemaKeysSync(desiredKey: SchemaKey, matchType: SchemaMatchType, format: string): FileSchemaKey[];
     // (undocumented)
     abstract getSchema<T extends Schema>(key: SchemaKey, matchType: SchemaMatchType, context: SchemaContext): Promise<T | undefined>;
     // (undocumented)
     protected abstract getSchemaKey(data: string): SchemaKey;
     // (undocumented)
+    getSchemaText(schemaPath: string): Promise<string | undefined>;
+    // (undocumented)
+    getSchemaTextSync(schemaPath: string): Promise<string | undefined> | undefined;
```

Comment:
it seems wrong that a sync method returns a promise

Reply:
I removed the sync method as the async method was the only one that was utilizing it internally, making the sync method unnecessary.